src/sniffer/pkg/collectors/tcpsniffer.go

```diff
@@ -0,0 +1,120 @@
+package collectors
+
+import (
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+	"github.com/google/gopacket/pcap"
+	"github.com/otterize/intents-operator/src/shared/errors"
+	sharedconfig "github.com/otterize/network-mapper/src/shared/config"
+	"github.com/otterize/network-mapper/src/sniffer/pkg/config"
+	"github.com/otterize/network-mapper/src/sniffer/pkg/ipresolver"
+	"github.com/otterize/nilable"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	"time"
+)
+
+type TCPSniffer struct {
+	NetworkCollector
+	resolver    ipresolver.IPResolver
+	pending     []pendingTCPCapture
+	lastRefresh time.Time
+}
+
+type pendingTCPCapture struct {
+	srcIp       string
+	srcHostname string
+	destIp      string
+	time        time.Time
+	ttl         nilable.Nilable[int]
+}
+
+func NewTCPSniffer(resolver ipresolver.IPResolver) *TCPSniffer {
+	s := TCPSniffer{
+		NetworkCollector: NetworkCollector{},
+		resolver:         resolver,
+		pending:          make([]pendingTCPCapture, 0),
+		lastRefresh:      time.Now().Add(-viper.GetDuration(config.HostsMappingRefreshIntervalKey)), // Should refresh immediately
+	}
+	s.resetData()
+	return &s
+}
+
+func setCaptureIncomingTCPSYN(handle *pcap.Handle) error {
+	err := handle.SetDirection(pcap.DirectionIn)
+	if err != nil {
+		return err
+	}
+	err = handle.SetBPFFilter("tcp and tcp[tcpflags] == tcp-syn")
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s *TCPSniffer) CreateTCPPacketStream() (chan gopacket.Packet, error) {
+	handle, err := pcap.OpenLive("any", 0, true, pcap.BlockForever)
+	if err != nil {
+		return nil, err
+	}
+
+	err = setCaptureIncomingTCPSYN(handle)
+	if err != nil {
+		return nil, err
+	}
+
+	packetSource := gopacket.NewPacketSource(handle, handle.LinkType())
+	return packetSource.Packets(), nil
+}
+
+func (s *TCPSniffer) HandlePacket(packet gopacket.Packet) {
+	if !viper.GetBool(sharedconfig.EnableTCPKey) {
+		return
+	}
+	captureTime := detectCaptureTime(packet)
+	ipLayer := packet.Layer(layers.LayerTypeIPv4)
+	if ipLayer != nil {
+		ip, _ := ipLayer.(*layers.IPv4)
```

Comment:
Check for ok here or is it always ipv4 ?

Reply:
I guess returning && printing debug log won't hurt